sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/RowWriterFactory.java

```diff
@@ -129,7 +127,18 @@ OutputType getOutputType() {
     @Override
     BigQueryRowWriter<ElementT> createRowWriter(String tempFilePrefix, DestinationT destination)
         throws Exception {
+      if (dynamicDestinations == null) {
+        throw new IllegalStateException(
+            "createRowWriter called when dynamicDestinations is null; forgot to call prepare()?");
+      }
+      if (schemaFactory == null) {
+        throw new IllegalStateException(
+            "createRowWriter called when schemaFactory is null; forgot to call prepare()?");
+      }
       TableSchema tableSchema = dynamicDestinations.getSchema(destination);
+      if (tableSchema == null) {
```

Comment:
This looks like a regression, in the past it was legal for `tableSchema` to be null here, as long as `schemaFactory` didn't require it.  I'm going to submit a PR to revert this specific null check.

Reply:
SGTM. You'll need to add the requirement that all `schemaFactory` that can be passed in can handle a null.